tests/unit/test_interaction.py

```diff
@@ -165,3 +213,118 @@ def test_interaction_matches_request_negative(
+
+
+@mark.parametrize(
+    'interaction_dir_files, expected_next_sequence_number',
+    [
+        (['interaction_1.yaml', 'interaction_0.yaml'], 2),
+        (['interaction_0.yaml'], 1),
+        ([], 0),
+        (['interaction_7.yaml', 'interaction_13.yaml'], 14),
+        (['interaction_1.yaml', 'other.yaml', 'interaction_0.yaml', 'another.yaml'], 2),
+        (['interaction_3.yaml', 'other.yaml', 'interaction_5.yaml', 'another.yaml'], 6),
+        (['interaction_1.yaml', 'interaction_0.yaml', 'interaction_2.zip'], 2),
+        (['.DS_Store', 'interaction_0.yaml~', 'something.jpeg', 'something.tar.gz'], 0),
+    ],
+    ids=(
+        'two interactions, continuous sequence from 0 => 2',
+        'one interaction, from 0 => 1',
+        'empty directory => 0',
+        'two interactions, not continuous (7, 13) => 14',
+        'two sequential interactions, two named otherwise, continuous sequence from 0 => 2',
+        'two sequential interactions, two named otherwise, not continuous (3, 5) => 6',
+        'two interactions, other file present (discard), continuous sequence from 0 => 2',
+        'directory not empty, but all files irrelevant => 0',
+    )
+)
+def test_interaction_get_next_sequence_number_in_dir(
+    tmpdir,
+    interaction_dir_files: List[str],
+    expected_next_sequence_number: int,
+):
+    interaction_dir = tmpdir.mkdir('interactions')
+    for file_name in interaction_dir_files:
+        interaction_dir.join(file_name).ensure()
+
+    interaction_dir_path = Path(interaction_dir.strpath)
+
+    result = Interaction.get_next_sequence_number_in_dir(interaction_dir_path)
+
+    assert expected_next_sequence_number == result
+
+
+@mark.parametrize(
+    'interaction_dir_files, expected_next_interaction_name',
+    [
+        (['interaction_1.yaml', 'interaction_0.yaml'], 'interaction_2'),
+        (['interaction_0.yaml'], 'interaction_1'),
+        ([], 'interaction_0'),
+    ],
+    ids=(
+        'two interactions, continuous sequence from 0 => 2',
+        'one interaction, from 0 => 1',
+        'empty directory => 0',
+    )
+)
+def test_interaction_next_in_dir(
+    tmpdir,
+    interaction_dir_files: List[str],
+    expected_next_interaction_name: str,
+):
+    interaction_dir = tmpdir.mkdir('interactions')
+    for file_name in interaction_dir_files:
+        interaction_dir.join(file_name).ensure()
+
+    interaction_dir_path = Path(interaction_dir.strpath)
+
+    result = Interaction.next_in_dir(
+        directory=interaction_dir_path,
+        request=MITMRequest(url='http://test.local/', method='GET'),
+        response=MITMResponse(status_code=204)
+    )
+
+    expected_request = MITMRequest(url='http://test.local/', method='GET')
+    expected_response = MITMResponse(status_code=204)
+
+    assert expected_next_interaction_name == result.name
+    assert expected_request == result.request
+    assert expected_response == result.response
+
+
+def test_save_in_dir(tmpdir, interaction_only_required_fields):
```

Comment:
This test is testing the wrong stuff, in this case too much (as it's meant to be a unit test), that is already tested in the integration tests.